akka-stream-tests/src/test/scala/akka/stream/scaladsl/SinkSpec.scala

```diff
@@ -324,6 +324,18 @@ class SinkSpec extends StreamSpec with DefaultTimeout with ScalaFutures {
     }
   }
 
+  "The reduce sink" must {
+    "sum up 1 to 10 correctly" in {
+      //#reduce-operator-example
+      val source = Source(1 to 10)
+      val rightValue = source.toMat(Sink.reduce((a: Int, b: Int) ⇒ a + b))(Keep.right).run
```

Comment:
If you put an explicit type param on reduce instead you can do `Sink.reduce[Int]((a, b) => a + b), and then use `runWith(sink)` it returns the materialized value from the sink without having to use the `Keep.right` part (last part is true for Java as well).

Reply:
done.